File: DSA/Array/SquaresOfSortedArray/Solution.1.js
// 977. Squares of a Sorted Array => https://leetcode.com/problems/squares-of-a-sorted-array/description/

const sorted = function (arr) {
    for (let i = 0; i < arr.length; i++) {
        arr[i] = arr[i] ** 2
    }

    arr.sort((x, y) => x - y)
    return arr
}

let sortArray = [-5, -4, -3, -2, 3, 4, 5, 6]

sorted(sortArray)
console.log(sortArray);

File: DSA/DataStructures/Array/TwoSumTwo/Solution1.java
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/

package E_DSA.B_DataStructures.D_TwoSumTwo;

class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int i=0;
        int j=numbers.length-1;

        while(i<j) {
            if(numbers[i]+numbers[j]==target) {
                return new int[] {i+1, j+1};
            } else if(numbers[i]+numbers[j]>target) {
                j--;
            } else {
                i++;
            }
        }
        return new int[] {};
    }
}